src/chunkserver/watchdog/dog_keeper.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2020-12-11
+ * Author: qinyi
+ */
+
+#include "src/chunkserver/watchdog/dog_keeper.h"
+
+#include <glog/logging.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+#include "src/chunkserver/watchdog/common.h"
+
+namespace curve {
+namespace chunkserver {
+
+DECLARE_bool(watchdogKillOnErr);
+DECLARE_bool(watchdogRevive);
+DECLARE_uint32(watchdogCheckPeriodSec);
+DECLARE_uint32(watchdogCheckTimeoutSec);
+DECLARE_uint32(watchdogLatencyCheckPeriodSec);
+DECLARE_uint32(watchdogLatencyCheckWindow);
+DECLARE_uint32(watchdogLatencyExcessMs);
+DECLARE_uint32(watchdogLatencyAbnormalMs);
+DECLARE_uint32(watchdogLatencyWindowSlowRatio);
+DECLARE_uint32(watchdogKillTimeoutSec);
+
+static void* Keeper(void* arg) {
+    DogKeeper* keeper = reinterpret_cast<DogKeeper*>(arg);
+    while (true) {
+        if (keeper->IsDestroy()) {
+            LOG(INFO) << "dogkeeper thread destroyed";
+            return nullptr;
+        }
+
+        keeper->UpdateConfig();
+
+        if (keeper->IsDogDead()) {
+            /* dog already dead, waiting for revive */
+            if (FLAGS_watchdogRevive) {
+                FLAGS_watchdogRevive = false;
+                keeper->ReviveDog();
+                LOG(INFO) << "watchdog revived, storDir="
+                          << keeper->conf_.storDir;
+            }
+            std::this_thread::sleep_for(
+                std::chrono::milliseconds(kWatchdogSleepMs));
+        } else {
+            keeper->FeedDog();
+            /* sleep 1 second */
```

Comment:
Inconsistent comments

Reply:
in fact, kWatchdogSleepFreq * kWatchdogSleepMs = 1000ms